safehttp/plugins/fetch_metadata/fetch_metadata_test.go

```diff
@@ -0,0 +1,399 @@
+				req.Header.Add("Sec-Fetch-Mode", "navigate")
+				req.Header.Add("Sec-Fetch-Dest", "image")
+				return req
+			}(),
+			wantStatus:  safehttp.StatusOK,
+			wantHeaders: map[string][]string{},
+			wantBody:    "",
+		},
+		{
+			name: "POST request with Sec-Fetch-Mode: navigate from image",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest("POST", "/", nil)
+				req.Header.Add("Sec-Fetch-Site", "cross-site")
+				req.Header.Add("Sec-Fetch-Mode", "navigate")
+				req.Header.Add("Sec-Fetch-Dest", "image")
+				return req
+			}(),
+			wantStatus: safehttp.StatusForbidden,
+			wantHeaders: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Forbidden\n",
+		},
+		{
+			name: "GET request with Sec-Fetch-Mode: navigate from object",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest("GET", "/", nil)
+				req.Header.Add("Sec-Fetch-Site", "cross-site")
+				req.Header.Add("Sec-Fetch-Mode", "navigate")
+				req.Header.Add("Sec-Fetch-Dest", "object")
+				return req
+			}(),
+			wantStatus: safehttp.StatusForbidden,
+			wantHeaders: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Forbidden\n",
+		},
+		{
+			name: "GET request with Sec-Fetch-Mode: navigate from embed",
+			req: func() *safehttp.IncomingRequest {
+				req := safehttptest.NewRequest("GET", "/", nil)
+				req.Header.Add("Sec-Fetch-Site", "cross-site")
+				req.Header.Add("Sec-Fetch-Mode", "navigate")
+				req.Header.Add("Sec-Fetch-Dest", "embed")
+				return req
+			}(),
+			wantStatus: safehttp.StatusForbidden,
+			wantHeaders: map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			},
+			wantBody: "Forbidden\n",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			p := fetchmetadata.NewPlugin()
+			rec := safehttptest.NewResponseRecorder()
+
+			p.Before(rec.ResponseWriter, test.req)
+			if status := safehttp.StatusCode(rec.Status()); status != test.wantStatus {
+				t.Errorf("status code got: %v want: %v", status, test.wantStatus)
+			}
+			if diff := cmp.Diff(test.wantHeaders, map[string][]string(rec.Header())); diff != "" {
+				t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+			}
+			if got := rec.Body(); got != test.wantBody {
+				t.Errorf("response body got: %q want: %q", got, test.wantBody)
+			}
+		})
+	}
+}
+
+type fooLog struct {
+	report string
+}
+
+func (l *fooLog) Log(r *safehttp.IncomingRequest) {
+	l.report = r.Method() + " " + r.URL.Path()
+}
+
+func TestResourceIsolationReportMode(t *testing.T) {
+	tests := []struct {
+		name        string
+		req         *safehttp.IncomingRequest
+		wantStatus  safehttp.StatusCode
+		wantHeaders map[string][]string
+		wantBody    string
```

Comment:
All of these tests have the same value for these parameters. Hardcode them in the subtest and remove them from the table.